public/app/plugins/panel/xychart/ManualEditor.tsx

```diff
@@ -101,23 +112,53 @@ export const ManualEditor = ({
       </div>
 
       {selected >= 0 && value[selected] && (
-        <ScatterSeriesEditor
-          key={`series/${selected}`}
-          baseNameMode={FieldNamePickerBaseNameMode.ExcludeBaseNames}
-          item={{} as StandardEditorsRegistryItem}
-          context={context}
-          value={value[selected]}
-          onChange={(v) => {
-            onChange(
-              value.map((obj, i) => {
-                if (i === selected) {
-                  return v!;
+        <>
+          {frameNames.length > 1 && (
+            <Field label={'Data'}>
+              <Select
+                isClearable={false}
+                options={frameNames}
+                placeholder={'Change filter'}
+                value={
+                  frameNames.find((v) => {
+                    return v.value === value[selected].frame;
+                  }) ?? 0
                 }
-                return obj;
-              })
-            );
-          }}
-        />
+                onChange={(v) => {
```

Comment:
Nit: I love when callback function args are typed, just for quick/glancing readability/context.